standard/statements.md

```diff
@@ -2018,6 +2054,101 @@ using (ResourceType rN = eN)
 >
 > *end example*
 
+A `using` statement of the form
+
+```csharp
+await using (ResourceType resource = «expression» ) «statement»
+```
+
+corresponds to the expansions shown above with `IAsyncDisposable` instead of `IDisposable`, `DisposeAsync` instead of `Dispose`, and the `Task` returned from `DisposeAsync` is `await`ed. An `await using` of the form:
+
+```csharp
+await using (ResourceType resource = «expression» ) «statement»
+```
+
+corresponds to
+
+```csharp
+{
+    ResourceType resource = «expression»;
+    try
+    {
+        «statement»;
+    }
+    finally
+    {
+        IAsyncDisposable d = (IAsyncDisposable)resource;
+        if (d != null)
+        {
+            await d.DisposeAsync();
+        }
+    }
+}
+```
+
+When `ResourceType` is a reference type that implements `IAsyncDisposable`. Other expansions for `await using` perform similar substitutions from the synchronous `Dispose` method to the asynchronous `DisposeAsync` method.
+
+> *Note*: Any jump statements (§13.10) in the *embedded_statement* must conform to expanded form of the `using` statement. *end note*
+
+### §using-declarations Using declaration
+
+A syntactic variant of the using statement is a *using declaration*.
+
+```ANTLR
+using_declaration
+    : 'await'? 'using' non_ref_local_variable_declaration ';' statement_list?
+    ;
+```
+
+A *using declaration* has the same semantics as, and can be rewritten as, the corresponding resource-acquisition form of the using statement (§using-general), as follows:
+
+```csharp
+using «local_variable_type» «local_variable_declarators»
+// statements
+```
+
+is equivalent to
+
+```csharp
+using («local_variable_type» «local_variable_declarators»)
+{
+    // statements
+}
+```
+
+and
+
+```csharp
+await using «local_variable_type» «local_variable_declarators»
+// statements
+```
+
+is equivalent to
+
+```csharp
+await using («local_variable_type» «local_variable_declarators»)
+{
+    // statements
+}
+```
+
+The lifetime of the variables declared in a *local_variable_declaration* extends to the end of the scope in which they are declared. Those variables are then disposed in the reverse order in which they are declared.
```

Comment:
Should *local_variable_declaration* here be *non_ref_local_variable_declaration*?